packages/twenty-server/src/engine/twenty-orm/factories/workspace-datasource.factory.ts

```diff
@@ -0,0 +1,58 @@
+import { Inject, Injectable, Scope } from '@nestjs/common';
+import { REQUEST } from '@nestjs/core';
+
+import { DataSource, EntitySchema } from 'typeorm';
+
+import { DataSourceService } from 'src/engine/metadata-modules/data-source/data-source.service';
+import { EnvironmentService } from 'src/engine/integrations/environment/environment.service';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { DataSourceStorage } from 'src/engine/twenty-orm/storage/data-source.storage';
+
+@Injectable({ scope: Scope.REQUEST })
```

Comment:
DataSource it automatically created based on each request, a storage is system is in place to avoid creating too many data sources.